src/cli.rs

```diff
@@ -631,53 +691,210 @@ fn construct_url(
+    //   request-item-2
+    //     help help
+    //
+    // normal help normal help
+    // ```
+    //
+    // Should look like this with roff controls
+    //
+    // ```
+    // normal help normal help
+    // normal help normal help
+    // .RS 12
+    // .TP
+    // request-item-1
+    // help help
+    // .TP
+    // request-item-2
+    // help help
+    // .RE
+    //
+    // .RS
+    // normal help normal help
+    // .RE
+    // ```
+    let lines: Vec<&str> = request_items_help.lines().collect();
+    let mut rs = false;
+    for i in 0..lines.len() {
+        if lines[i].is_empty() {
+            let prev = lines[i - 1].chars().take_while(|&x| x == ' ').count();
+            let next = lines[i + 1].chars().take_while(|&x| x == ' ').count();
+            if prev != next && next > 0 {
+                if !rs {
+                    request_items_roff.control("RS", ["8"]);
+                    rs = true;
+                }
+                request_items_roff.control("TP", ["4"]);
+            } else if prev != next && next == 0 {
+                request_items_roff.control("RE", []);
+                request_items_roff.text(vec![roman("")]);
+                request_items_roff.control("RS", []);
+            } else {
+                request_items_roff.text(vec![roman(lines[i])]);
+            }
+        } else {
+            request_items_roff.text(vec![roman(lines[i].trim())]);
+        }
+    }
+    request_items_roff.control("RE", []);
+
+    let mut options_roff = Roff::new();
+    let non_pos_items = items
+        .iter()
+        .filter(|a| !a.is_positional())
+        .collect::<Vec<_>>();
+    // move the first two items (i.e. --help, --version) to the end
+    let non_pos_items = non_pos_items
+        .iter()
+        .cycle()
+        .skip(2)
+        .take(non_pos_items.len());
+
+    for opt in non_pos_items {
+        let mut header = vec![];
+        if let Some(short) = opt.get_short() {
+            header.push(bold(format!("-{}", short)));
+        }
+        if let Some(long) = opt.get_long() {
+            if !header.is_empty() {
+                header.push(roman(", "));
+            }
+            header.push(bold(format!("--{}", long)));
+        }
+        if let Some(value) = &opt.get_value_names() {
+            if opt.get_long().is_some() {
+                header.push(roman("="));
+            } else {
+                header.push(roman(" "));
+            }
+
+            if opt.get_id() == "auth" {
+                header.push(italic("USER"));
+                header.push(roman("["));
+                header.push(italic(":PASS"));
+                header.push(roman("] | "));
+                header.push(italic("TOKEN"));
+            } else {
+                header.push(italic(&value.join(" ")));
+            }
+        }
+        let mut body = vec![];
+
+        let help = opt
+            .get_long_help()
+            .or_else(|| opt.get_help())
+            .expect("option is missing help");
```

Comment:
```suggestion
        let mut help = opt
            .get_long_help()
            .or_else(|| opt.get_help())
            .expect("option is missing help")
            .to_owned();
        if !help.ends_with('.') {
            help.push('.')
        }
```
This gives a more consistent style. I've found manpages that always end an option description with a period, and ones that never end it with a period and always start it with a lowercase letter, but I've found none that mix periods like clap does.

(Maybe we can change this in clap instead? I wouldn't mind seeing more periods in the long help. But that's something to look at once we're on clap 4.)